src/crimson/os/seastore/seastore_types.h

```diff
@@ -36,29 +36,139 @@ struct seastore_meta_t {
 };
 
 // Identifies segment location on disk, see SegmentManager,
-using segment_id_t = uint32_t;
+using internal_segment_id_t = uint32_t;
+using device_id_t = uint8_t;
+constexpr uint16_t device_id_len = 4;
+constexpr uint16_t max_devices = 1 << device_id_len;
+
+using device_segment_id_t = internal_segment_id_t;
+
+// mask for segment manager id
+constexpr device_segment_id_t SM_ID_MASK =
+  0xF << (std::numeric_limits<device_segment_id_t>::digits - device_id_len);
+
+struct segment_id_t {
+  device_segment_id_t segment;
```

Comment:
Please audit this PR for uninitialized member variables like this one.  device_segment_id_t = internal_segment_id_t = uint32_t which does not have a default initializer, so your default constructor below will leave it uninitialized.

Reply:
I added internal_segment_id_t to represent a segment id without device id, I thought this might be useful when we are referencing segment ids within the single segment manager scope. Am I right?

Reply:
This comment referred specifically to the fact that it will have undefined contents after the default constructor is called.  Change the default constructor to set itself to NULL_SEG_ID.  There aren't really any cases where it's ok to leave a variable uninitialized.

Reply:
Yes, sorry that I left the comment at the wrong place, it should have been added [here](https://github.com/ceph/ceph/pull/43181#discussion_r712669743)